pkg/controllers/teamrbac/teamrolebinding_controller.go

```diff
@@ -564,10 +572,17 @@ func reconcileRoleBinding(ctx context.Context, cl client.Client, c *greenhousev1
 		remoteRB.Subjects = rb.Subjects
 		return nil
 	})
-
 	if err != nil {
-		return err
+		if createNamespaces && apierrors.IsNotFound(err) {
+			err := cl.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: rb.Namespace}})
+			if err != nil {
+				return err
+			}
```

Comment:
An error has to be returned in every case to force the reconciliation to be triggered again.
```suggestion
			err := cl.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: rb.Namespace}})
			if err != nil {
				return err
			}
			return fmt.Errorf("failed to create RoleBinding, created missing namespace")
```